### PR TITLE
Add MonthParameter and YearParameter

### PR DESCRIPTION
**Pull request by soxofaan**
Add MonthParameter and YearParameter

### CODE REVIEW

**Comment by erikbern** [comment]
weird i thought we already had that :)

**Comment by erikbern** [comment]
oh right we have the underlying `DateInterval` types but not the corresponding `Parameter` class

File: test/date_parameter_test.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    def test_parse_interface(self):
+        task = luigi.interface.ArgParseInterface().parse(["DateMinuteTask", "--dm", "2013-02-01T18H42"])[0]
+        self.assertEqual(task.dm, datetime.datetime(2013, 2, 1, 18, 42, 0))
+
+    def test_serialize_task(self):
+        t = DateMinuteTask(datetime.datetime(2013, 2, 1, 18, 42, 0))
+        self.assertEqual(str(t), 'DateMinuteTask(dm=2013-02-01T18H42)')
+
+
+class MonthTest(unittest.TestCase):
```
Comment (Tarrasch) [inline test/date_parameter_test.py]: Either stick with the "DateBlahTest" convention or change the other examples. :)

**Comment by Tarrasch** [comment]
Thanks for the patch. Sorry for slow reply. Mind just renaming the test classes? I think this is good to merge otherwise. :)

**Comment by soxofaan** [comment]
@Tarrasch : well, I chose `MonthParameter` and `YearParameter` (next to `DateParameter`, `DateHourParameter`, `DateMinuteParameter`), because something like `DateMonthParameter` seemed like needlessly redundant.
Likewise, I named the test classes `MonthTest` and `YearTest` and `DateTest` (next to existing `DateHourTest` and DateMinuteTest`)

Do you want me to change the former too (to `DateMonthParameter` and `DateYearParameter`)?

**Comment by Tarrasch** [comment]
Ah, it was already mixed then. Nevermind then. LGTM. Thanks! :)

**Comment by Tarrasch** [comment]
That said, it would of course be awesome if you renamed stuff to be consistent in the test file. :)

**Pull request merged**